omnigan/eval_metrics.py

```diff
@@ -63,4 +65,39 @@ def f1_score(pred_im, gt_im):
 def accuracy(pred_im, gt_im):
     pred = np.array(pred_im)
     gt = np.array(gt_im)
+    if len(pred.shape) > len(gt_im.shape):
+        pred = np.argmax(pred, axis=1)
     return float((pred == gt).sum()) / gt.size
+
+
+def mIOU(label, pred, num_classes=19):
+    """
+    https://stackoverflow.com/questions/62461379/multiclass-semantic-segmentation-model-evaluation
+    """
+    pred = F.softmax(pred, dim=1)
+    pred = torch.argmax(pred, dim=1).squeeze(1)
+    iou_list = list()
+    present_iou_list = list()
+
+    num_classes = pred.shape[-3]
```

Comment:
why put a num_classes param if you reset it here ?

Reply:
Also I am not sure this gives the right number of classes?

Reply:
num_params is legacy from where I got the code

Reply:
the number of classes is definitely the number of channels of the predicion **but** you're right that there's an error here! num_classes should be defined before the argmax!

Reply:
That's what I thought :)

Reply:
It seems good now !